src/averageAges.js

```diff
@@ -36,8 +40,18 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  let women = people.filter(person => person.sex === 'f');
+
+  women = withChildren === false
+    ? women
+    : women
+      .filter(person => people
+        .some(child => child.mother === person.name));
+
+  const ages = women.map(person => person.died - person.born);
+
+  return ages.reduce((x, y) => x + y, 0) / ages.length;
```

Comment:
the same, check your function above and my suggestion for some callback

follow DRY principle (Don't Repeat Yourself)